File: frontend/src/types/Order.ts
export type OrderStatus =
  | "PENDING"
  | "PAID"
  | "SHIPPED"
  | "DELIVERED"
  | "CANCELLED"

export type OrderItemRequest = {
  productId: number
  quantity: number
}

export type OrderItemResponse = {
  productId: number
  productName: string
  quantity: number
  priceAtPurchase: number
  subtotal: number
}

export type CreateOrderRequest = {
  deliveryFullName: string
  deliveryPhone: string
  deliveryAddress: string
  deliveryCity: string
  deliveryPostalCode: string
  deliveryNotes: string
  items: OrderItemRequest[]
}

export type OrderResponse = {
  id: number
  totalPrice: number
  status: OrderStatus
  createdAt: string

  userId: number | null
  customerName: string | null
  customerEmail: string | null

  deliveryFullName: string | null
  deliveryPhone: string | null
  deliveryAddress: string | null
  deliveryCity: string | null
  deliveryPostalCode: string | null
  deliveryNotes: string | null

  items: OrderItemResponse[]
}

export type Order = OrderResponse

File: backend/src/main/java/com/wardro/backend/order/Order.java
package com.wardro.backend.order;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import com.wardro.backend.auth.AppUser;


@Entity
@Table(name = "orders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private BigDecimal totalPrice;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<OrderItem> items = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "user_id")
    private AppUser user;

    private String deliveryFullName;

    private String deliveryPhone;

    private String deliveryAddress;

    private String deliveryCity;

    private String deliveryPostalCode;

    private String deliveryNotes;

    @PrePersist
    public void onCreate() {
        this.createdAt = LocalDateTime.now();

        if (this.status == null) {
            this.status = OrderStatus.PENDING;
        }
    }
}